```
--- FAIL: TestUpdatePidsLimit (4.45s)
    --- FAIL: TestUpdatePidsLimit/update_from_none (0.47s)
        update_linux_test.go:154: failed to parse source file: /go/src/github.com/docker/docker/integration/container/update_linux_test.go: open /go/src/github.com/docker/docker/integration/container/update_linux_test.go: no such file or directory
```

(most probably the failure comes because I don't run it as root, I don't care)

It looks like test case sources should be copied to the final image, because stuff like `assert.Equal()` etc tries to parse (reflect) the source file in case of failure to print info about what kind of variable and what value is expected and what is provided, stuff like that.

--- a/integration/container/ipcmode_linux_test.go
+++ b/integration/container/ipcmode_linux_test.go
@@ -300,12 +300,14 @@ func TestDaemonIpcModeShareableFromConfig(t *testing.T) {
 // by default, even when the daemon default is private.
 func TestIpcModeOlderClient(t *testing.T) {
 	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.40"), "requires a daemon with DefaultIpcMode: private")
+	c := testEnv.APIClient()
+	skip.If(t, versions.LessThan(c.ClientVersion(), "1.40"), "requires client API >= 1.40")
+
 	t.Parallel()
 
 	ctx := context.Background()
 
 	// pre-check: default ipc mode in daemon is private
-	c := testEnv.APIClient()
 	cID := container.Create(t, ctx, c, container.WithAutoRemove)
 
 	inspect, err := c.ContainerInspect(ctx, cID)

